container/new_pull.bzl

```diff
@@ -0,0 +1,208 @@
+    repository_ctx.file("image/BUILD", """
+package(default_visibility = ["//visibility:public"])
+
+load("@io_bazel_rules_docker//container:import.bzl", "container_import")
+
+container_import(
+    name = "image",
+    config = "image/index.json",
+    layers = glob(["image/blobs/sha256/*"]),
+)
+    """)
+
+    args = [
+        repository_ctx.path(repository_ctx.attr._puller),
+        "-directory",
+        repository_ctx.path("image"),
+        "-os",
+        repository_ctx.attr.os,
+        "-os-version",
+        repository_ctx.attr.os_version,
+        "-os-features",
+        " ".join(repository_ctx.attr.os_features),
+        "-architecture",
+        repository_ctx.attr.architecture,
+        "-variant",
+        repository_ctx.attr.cpu_variant,
+        "-features",
+        " ".join(repository_ctx.attr.platform_features),
+    ]
+
+    # Use the custom docker client config directory if specified.
+    if repository_ctx.attr.docker_client_config != "":
+        args += ["-client-config-dir", "{}".format(repository_ctx.attr.docker_client_config)]
+
+    # TODO(xwinxu): There is currently no support for caches in rules_docker. Stay tuned...
+    # cache_dir = repository_ctx.os.environ.get("DOCKER_REPO_CACHE")
+    # if cache_dir:
+    #     if cache_dir.startswith("~/") and "HOME" in repository_ctx.os.environ:
+    #         cache_dir = cache_dir.replace("~", repository_ctx.os.environ["HOME"], 1)
+
+    #     args += [
+    #         "-cache",
+    #         cache_dir,
+    #     ]
+
+    # If a digest is specified, then pull by digest.  Otherwise, pull by tag.
+    if repository_ctx.attr.digest:
+        args += [
+            "-name",
+            "{registry}/{repository}@{digest}".format(
+                registry = repository_ctx.attr.registry,
+                repository = repository_ctx.attr.repository,
+                digest = repository_ctx.attr.digest,
+            ),
+        ]
+    else:
+        tag_ = repository_ctx.attr.tag
+        if not repository_ctx.attr.tag:
+            tag_ = "latest"
+        args += [
+            "-name",
+            "{registry}/{repository}:{tag}".format(
+                registry = repository_ctx.attr.registry,
+                repository = repository_ctx.attr.repository,
+                tag = tag_,
+            ),
+        ]
+
+    kwargs = {}
+    if "PULLER_TIMEOUT" in repository_ctx.os.environ:
+        kwargs["timeout"] = int(repository_ctx.os.environ.get("PULLER_TIMEOUT"))
+
+    result = repository_ctx.execute(args, **kwargs)
+    if result.return_code:
+        fail("Pull command failed: %s (%s)" % (result.stderr, " ".join(args)))
+
+    updated_attrs = {
+        k: getattr(repository_ctx.attr, k)
+        for k in _container_pull_attrs.keys()
+    }
+    updated_attrs["name"] = repository_ctx.name
+
+    # TODO(xwinxu): if repository_ctx.attr.digest is supported by puller (currently only outputting OCI Layout, cannot parse json, 
```

Comment:
I think the purpose of the following piece old in the old pull.bzl is to

1: Return the resolved digest to bazel (https://github.com/bazelbuild/rules_docker/commit/e5785ceaef4eb7e0cc28bdb909fd1b10d5b991c3) so it can be used by bazel workspace resolve file
2. Fail when the pulled image digest is different from the digest user specifies (https://github.com/bazelbuild/rules_docker/pull/734/files).

We should be able to support them as well. Can you check if there is any file in the OCI layout pulled down that contains the digest of the image in the registry, or contains the same value of `image/digest`  that the python puller would pull down?

Reply:
Yes we have index.json, which is an entrypoint and lists the manifests of the images in the layout pulled. We would have to take the code in index.json and go to the corresponding file in blobs/sha256 to get the same digest as image/digest pulled by python puller.

Reply:
But we don't have a file like the python where the content is just the digest.